### PR TITLE
SASS-1651 - Create taxable employment benefits amount EOY page.

### PR DESCRIPTION
**Pull request by georgievh**
### Description
Creates taxable employment benefits amount EOY page.

Add a link to the relevant story in Jira
[SASS-1651](https://jira.tools.tax.service.gov.uk/browse/SASS-1651)

##### Before Reviewing
- [x]  Have you pulled the branch down?
- [x]  Have you assigned yourself to the PR?
- [x]  Have you moved the task to “in review” on JIRA?
- [x]  Have you checked to ensure all dependencies are up to date?
- [x]  Have you checked to ensure its been rebased against the current version of main?

##### Whilst Reviewing
- [x]  Have you run the tests?
- [x]  Have you run the journey tests?
- [x]  Have you looked at the JIRA story to make sure all Acceptance Criteria has been met?

##### After Reviewing
- [ ]  Have you checked for merge conflicts or any changes in the current main that may affect the current pull request? i.e. does it need another rebase?
- [ ]  Have you checked to make sure there are no builds in the pipeline before you merge?
- [ ]  Have you moved the task to “in pipeline” on Jira?

##### Before creating PR
- [x]  Have you run the tests?
- [x]  Have you run the journey tests? (where applicable)
- [x]  Have you addressed warnings where appropriate?
- [x]  Have you rebased against the current version of main?
- [x]  Have you checked code coverage isn’t lower than previously?

##### After PRs been raised
- [x]  Have you checked the PR Builder passes?

### CODE REVIEW

File: scalastyle-test-config.xml [hunk 1]
```diff
@@ -30,7 +30,7 @@
     <check level="warning" class="org.scalastyle.scalariform.SpacesBeforePlusChecker" enabled="true"></check>
     <check level="warning" class="org.scalastyle.file.FileLineLengthChecker" enabled="true">
         <parameters>
-            <parameter name="maxLineLength"><![CDATA[200]]></parameter>
```
Comment (joeyymoore) [inline scalastyle-test-config.xml:33]: is it good to change this? should we keep it low to discourage people doing long lines of code?

**Review by joeyymoore — CHANGES_REQUESTED** [review]

File: it/controllers/benefits/reimbursed/TaxableCostsBenefitsAmountControllerISpec.scala [hunk 2]
```diff
@@ -0,0 +1,465 @@
...
+  private val employmentId = "001"
+  private val userRequest = User(mtditid, None, nino, sessionId, affinityGroup)(fakeRequest)
+  private val continueLink = s"/update-and-submit-income-tax-return/employment-income/$taxYearEOY/benefits/taxable-costs-amount?employmentId=$employmentId"
+
+  private def pageUrl(taxYear: Int): String = s"$appUrl/$taxYear/benefits/taxable-costs-amount?employmentId=$employmentId"
+
+  private def employmentUserData(isPrior: Boolean, employmentCyaModel: EmploymentCYAModel): EmploymentUserData =
+    EmploymentUserData(sessionId, mtditid, nino, taxYearEOY, employmentId, isPriorSubmission = isPrior, hasPriorBenefits = isPrior, employmentCyaModel)
+
+  def cyaModel(hmrc: Boolean, benefits: Option[BenefitsViewModel] = None): EmploymentCYAModel =
```
[the remark this answers is not in the document]
Comment (georgievh) [reply it/controllers/benefits/reimbursed/TaxableCostsBenefitsAmountControllerISpec.scala:42]: Ah, I have noticed that and forgot to change it. Will do that now. Thank you

File: scalastyle-test-config.xml (hunk 1 above)
Comment (georgievh) [reply scalastyle-test-config.xml:33]: In production this is still low. We have changed it once with Ben to 200, so I thought I could increase it again, as this is for tests only. We should address the warnings one way or another. We can check with Ben or someone else to see what we should do.

Comment (BenCosfordHMRC) [reply scalastyle-test-config.xml:33]: I didn't mind the increase to 200 but I agree with Joey we should still keep it relatively similar to the production file so the tests stay rather readable.

Comment (TheDeadlyPianist) [reply scalastyle-test-config.xml:33]: I'm with joey on this one. I don't see much point in raising it higher.

Comment (georgievh) [reply scalastyle-test-config.xml:33]: Sure, I will put it back to 200. We have to resolve the warnings though. Probably will be part of another work.

File: it/controllers/benefits/reimbursed/TaxableCostsBenefitsAmountControllerISpec.scala [hunk 3]
```diff
@@ -0,0 +1,465 @@
...
+          insertCyaData(employmentUserData(isPrior = false, cyaModel(hmrc = true, benefitsWithNoBenefitsReceived)), userRequest)
+          authoriseAgentOrIndividual(isAgent = false)
+          urlGet(pageUrl(taxYear), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)))
+        }
+
+        result.status shouldBe SEE_OTHER
+        result.header("location") shouldBe Some(appConfig.incomeTaxSubmissionOverviewUrl(taxYear))
+      }
+
+      "Redirect to 'Vouchers, credit cards or mileage allowance' page when there is taxableExpenses amount but has no taxableExpensesQuestion" in {
```
[the remark this answers is not in the document]
Comment (georgievh) [reply it/controllers/benefits/reimbursed/TaxableCostsBenefitsAmountControllerISpec.scala:264]: Yes, I will change the name. Good spot. Thank you

File: it/controllers/benefits/reimbursed/TaxableCostsBenefitsAmountControllerISpec.scala [hunk 4]
```diff
@@ -0,0 +1,465 @@
...
+      "redirect to taxable expenses question page when there is taxableExpenses amount but has no taxableExpensesQuestion" in {
+        implicit lazy val result: WSResponse = {
+          authoriseAgentOrIndividual(isAgent = false)
+          dropEmploymentDB()
+          val model = fullReimbursedCostsVouchersAndNonCashModel.copy(taxableExpensesQuestion = None)
+          insertCyaData(employmentUserData(isPrior = false, cyaModel(hmrc = true, Some(benefits(model)))), userRequest)
+          urlPost(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)), body = Map("amount" -> "100"))
+        }
+
+        result.status shouldBe SEE_OTHER
```
[the remark this answers is not in the document]
Comment (georgievh) [reply it/controllers/benefits/reimbursed/TaxableCostsBenefitsAmountControllerISpec.scala:460]: Done

File: it/controllers/benefits/reimbursed/TaxableCostsBenefitsAmountControllerISpec.scala [hunk 5]
```diff
@@ -0,0 +1,465 @@
...
+
+      "redirect to to Vouchers yes/no page when a valid form is submitted and not prior submission" when {
+        implicit lazy val result: WSResponse = {
+          authoriseAgentOrIndividual(isAgent = false)
+          dropEmploymentDB()
+          insertCyaData(employmentUserData(isPrior = false, cyaModel(hmrc = true, Some(benefits(fullReimbursedCostsVouchersAndNonCashModel)))), userRequest)
+          urlPost(pageUrl(taxYearEOY), follow = false, headers = Seq(HeaderNames.COOKIE -> playSessionCookies(taxYearEOY)), body = Map("amount" -> "234.56"))
+        }
+
+        "has an SEE_OTHER status" in {
```
[the remark this answers is not in the document]
Comment (georgievh) [reply it/controllers/benefits/reimbursed/TaxableCostsBenefitsAmountControllerISpec.scala:411]: Done

**Review by joeyymoore — APPROVED** [review]

**Pull request merged**